### PR TITLE
[Docs] upgrade steps and best practices

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Creates a topic that provides guidance and best practices for upgrading a scaffolded project to the latest version.

This PR also adds a note to the CHANGELOG to add specific information to help the upgrade process in future release notes.

## Related Issue

Closes PWA-1819

## Acceptance

Any core developer
@davemacaulay 

### Verification Stakeholders

Any core developer
@davemacaulay 

## Test Plan

Docs change, so no test plan necessary.

#### Test scenario(s) for direct fix/feature
Example
-   Verify user is able to apply filters on category page
-   Verify user is able to apply filters on search page

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/technologies/upgrading-versions/index.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+You can also use one of the many [diff tools in Visual Studio Code][] or a standalone application, such as [GitHub's desktop application][], to make these comparisons easier.
+
+[diff tools in Visual Studio Code]: https://marketplace.visualstudio.com/search?term=diff&target=VSCode&category=All%20categories&sortBy=Relevance
+[github's desktop application]: https://desktop.github.com/
+
+## Check for new environment variables
+
+To check for new or updated environment variables, run a diff on the `envVarDefinitions.json` file in the **pwa-buildpack** package.
+For example, the following command shows the updates on that file between versions _v9.0.1_ and _v10.0.0_.
```
Comment (supernova-at) [inline pwa-devdocs/src/technologies/upgrading-versions/index.md:115]:
```suggestion
For example, the following command shows the updates on that file between versions `v9.0.1` and `v10.0.0`.
```

Minor update for consistency in this file 👍

**Review by supernova-at — CHANGES_REQUESTED** [review]
Looks great, I know this will be very valuable for folks.

Just a quick consistency change needed.

**Review by supernova-at — DISMISSED** [review]

File: pwa-devdocs/src/technologies/upgrading-versions/index.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+```sh
+yarn add @magento/eslint-config @magento/pagebuilder @magento/peregrine @magento/pwa-buildpack @magento/upward-js @magento/venia-ui
+```
+
+If your project uses `npm`, you can still manually edit the `package.json` file with the latest published version.
+Compare the contents of your project's `package.json` file with the one in `venia-concept` to see the latest published versions.
+For example, the `package.json` file for the **v10.0.0** release is at:
+
+<https://github.com/magento/pwa-studio/blob/v10.0.0/packages/venia-concept/package.json>
```
Comment (davemacaulay) [inline pwa-devdocs/src/technologies/upgrading-versions/index.md:34]: I feel like this comment should have more visibility as this is how we're informing them to find the version to upgrade.

Could we maybe include a short example of finding the new version, modifying it in your project and running an update?

File: pwa-devdocs/src/technologies/upgrading-versions/index.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## Update template files
+
+Updating your dependencies does not update the initial template files in your project.
+You must identify and apply these changes manually so that they do not conflict with your own changes.
```
Comment (davemacaulay) [inline pwa-devdocs/src/technologies/upgrading-versions/index.md:39]: It'd be good to say we'll explore how to do this in the next section

**Review by davemacaulay — CHANGES_REQUESTED** [review]
Looks great overall, couple of minor tweaks

**Review by davemacaulay — APPROVED** [review]
